gengokit/gengokit.go

```diff
@@ -160,134 +265,64 @@ func (g *generator) updateClientMethods(clientHandler io.Reader, svcFuncs []stri
 	currentFuncs := astMod.IndexFunctions()
 	code := astMod.Buffer()
 
-	err = g.applyTemplateForMissingMeths(clientMethodsTemplPath, currentFuncs, code)
+	// Get a templateExecutor that is identical but with only functions in the
+	// definition file and not in the previously generated file
+	trimmedTe, err := trimTemplateExecutorServiceFuncs(*te, currentFuncs)
 
+	out, err := applyTemplateFromPath(clientMethodsTemplPath, trimmedTe)
 	if err != nil {
 		return nil, errors.Wrap(err, "unable to apply client methods template")
 	}
 
+	code.ReadFrom(out)
+
 	return code, nil
 }
 
-// GenerateResponseFiles applies all template files for the generated
-// microservice and returns a slice containing each templated file as a
-// truss.NamedReadWriter.
-func (g *generator) GenerateResponseFiles(previousFiles []truss.NamedReadWriter, packageName string) ([]truss.NamedReadWriter, error) {
-	// Paths to handler files that may be modified programmatically
-	serverHandlerFilePath := "-service/handlers/server/server_handler.go"
-	clientHandlerFilePath := "-service/handlers/client/client_handler.go"
-
-	var codeGenFiles []truss.NamedReadWriter
-
-	// serviceFunctions is used later as the master list of all service methods
-	// which should exist within the handler files
-	var serviceFunctions []string
-	for _, meth := range g.templateExec.Service.Methods {
-		serviceFunctions = append(serviceFunctions, meth.GetName())
+// serviceFunctionNames returns a slice of function names which are in the
+// definition files plus the function "NewService". Used for inserting and
+// removing functions from previously generated handler files
+func serviceFunctionsNames(meths []*deftree.ServiceMethod) []string {
+	var svcFuncs []string
+	for _, meth := range meths {
+		svcFuncs = append(svcFuncs, meth.GetName())
 	}
-	serviceFunctions = append(serviceFunctions, "NewBasicService")
-
-	serverHandlerFile := getFileByName(packageName+serverHandlerFilePath, previousFiles)
-	clientHandlerFile := getFileByName(packageName+clientHandlerFilePath, previousFiles)
-
-	for _, templateFilePath := range templateFileAssets.AssetNames() {
-		if filepath.Ext(templateFilePath) != ".gotemplate" {
-			log.WithField("Template file", templateFilePath).Debug("Skipping rendering non-buildable partial template")
-			continue
-		}
-
-		var generatedFilePath string
-		var generatedCode *bytes.Buffer
-		var err error
-
-		if templateFilePath == "NAME"+serverHandlerFilePath+"template" && serverHandlerFile != nil {
-			// If there's an existing service file, update its contents
+	svcFuncs = append(svcFuncs, "NewService")
```

Comment:
You're right, the explanation in the doc is sufficient, adding an additional comment like I suggested is unnecessary.